src/lib.rs

```diff
@@ -11,6 +11,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#![cfg_attr(test, deny(warnings))]
```

Comment:
It's ok. But I prefer put this to configuration file like `.travis.yml`. I usually fix the warnings at last (after all the tests passed).